test/second_q/operators/test_commutators.py

```diff
@@ -0,0 +1,80 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test for commutators"""
+
+from __future__ import annotations
+
+import unittest
+from test import QiskitNatureTestCase
+from ddt import ddt, data, unpack
+
+from qiskit_nature.second_q.operators import FermionicOp
+from qiskit_nature.second_q.operators.commutators import (
+    commutator,
+    anti_commutator,
+    double_commutator,
+)
+
+op1 = FermionicOp({"+_0 -_0": 1}, register_length=1)
+op2 = FermionicOp({"-_0 +_0": 2}, register_length=1)
+op3 = FermionicOp({"+_0 -_0": 1, "-_0 +_0": 2 + 0.5j}, register_length=1)
+op4 = FermionicOp({"+_0": 1}, register_length=1)
+op5 = FermionicOp({"-_0": 1}, register_length=1)
+
+
+@ddt
+class TestCommutators(QiskitNatureTestCase):
+    """Commutators tests."""
+
+    @unpack
+    @data(
+        (op1, op2, {}),
+        (op4, op5, {"+_0 -_0": (1 + 0j), "-_0 +_0": (-1 + 0j)}),
+    )
+    def test_commutator(self, op_a: FermionicOp, op_b: FermionicOp, expected: dict):
+        """Test commutator method"""
+        self.assertEqual(commutator(op_a, op_b), FermionicOp(expected, register_length=1))
+
+    @unpack
+    @data(
+        (op1, op2, {}),
+        (op1, op3, {"+_0 -_0": (2 + 0j)}),
+    )
+    def test_anti_commutator(self, op_a: FermionicOp, op_b: FermionicOp, expected: dict):
+        """Test anti commutator method"""
+        self.assertEqual(anti_commutator(op_a, op_b), FermionicOp(expected, register_length=1))
+
+    @unpack
+    @data(
+        (op1, op2, op3, {}),
+        (op1, op4, op3, {"+_0": (1 + 0.5j)}),
+    )
```

Comment:
Please also add the `sign` as a parameter here and check the output when you change it